src/main/java/net/mrscauthd/beyond_earth/common/events/Events.java

```diff
@@ -236,4 +233,106 @@ public static void livingSprinting(LivingSprintingEvent event) {
             entity.level.playSound(null, entity, SoundRegistry.SONIC_BOOM_SOUND.get(), SoundSource.NEUTRAL, 1.0F, 1.0F);
         }
     }
+
+    @SubscribeEvent
+    public static void onDataSync(OnDatapackSyncEvent event) {
+        PlanetData data = new PlanetData();
+        Planets.ORDERED_STARS.forEach(s -> data.stars.add(new StarEntry(s)));
+        PlanetDataHandler holder = new PlanetDataHandler();
+        holder.data = data;
+        NetworkRegistry.PACKET_HANDLER.send(PacketDistributor.PLAYER.with(() -> event.getPlayer()), holder);
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    /**
+     * Here we register default planets. We are set to HIGHEST so that we fire
+     * first, and then addons can adjust things.
+     * 
+     * @param event
+     */
+    public static void onPlanetsGenerate(PlanetRegisterEvent.Generate event) {
```

Comment:
@gisellevonbingen I added 2 events, `PlanetRegisterEvent#Generate` and `PlanetRegisterEvent#Load`, the base mod will use the former to add planets, as done here, addons can either use this event, or `PlanetRegisterEvent#Load` to also add their planets, the way it is done here should work as an example for both cases

Reply:
PlanetSelectionScreenInitEvent is looks coarse
Branching on a 'category ID' integer variable is very outdated
Minecraft also changed block/item ID from integer to ResourceLocation

Best in my think is 'beyond earth'  be managing planet's relationship.
If register start system, planet and moons data in config.json or other ways,
It have to be show automatically in PlanetSelectionGUI

just my thoughts
If you think you don't have to, that's fine.

Reply:
I didn't make any changes to that, since that was part of the older API code.

My changes were to the internal, non-addon facing parts of the planet system. I did remove the giant wall of switch-case statements for the IDs though.

In the case for internal use however, the "category ID" is fine though, as that is a sequential system anyway, as it loops through the stars -> planets in the correct order

Reply:
OK, i had look code some more,
Most of them seem to be
https://github.com/MrScautHD/Beyond-Earth/pull/185/files#diff-53634af117cdb210e31fb45406626b30d5a4a14bbb459c5114e8a50571456be3R240-R260

Reply:
that link seems to be corrupted, it just takes me to the top of the diffs, which file?

Reply:
Ah never mind
I saw it wrong
https://github.com/MrScautHD/Beyond-Earth/blob/003ed4b980d2dc200fdc61057b88a9fc12be8fa9/src/main/java/net/mrscauthd/beyond_earth/client/screens/planetselection/PlanetSelectionScreen.java#L242-L260

Reply:
ya, they start at 0, then increase as you go to planets, it doesn't increase for moons/orbits.

This allows a reasonable way to maintaining the tree for finding which button should do what. It is also not user-facing, so numeric IDs are fine, and the numeric IDs can be assigned easily and automatically